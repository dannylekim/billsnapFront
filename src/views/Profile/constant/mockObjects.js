export const mockUserInfo = { 
    firstName: "string",
    middleName: "string",
    lastName: "string",
    phoneNumber: "string",
    birthDate : "string",
    location: {city: "string",country: "string"},
    email: "string"
};