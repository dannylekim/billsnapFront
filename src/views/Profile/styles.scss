@import "../../variables";

div.profile__section{
    width:100%;
    height:100%;
    background-color: white;
    
    div.header__profile{
        background-color: rgb(148, 230, 182);
        width:inherit;
        height:26.3%;
    }

    img.profile__avatar{
        margin-top:-8%;
        border-radius: 100px;
    }

    div.user__name{
        margin-top: 2%;
        display: flex;
        width:500px;
        margin-left: auto;
        margin-right: auto;
        h1{
            text-align: center;
        }

        button.edit__profile{
            width: 55px;
            margin-left: auto;
            background: $secondary-color;
            border-radius: 13px;
            color:$card-color;
            border: none;
        }
    }

    div.profile__info{
        margin-top: 2%;
        height: 164px;
        width:75%;
        margin-left: auto;
        margin-right: auto;
        border: 1px solid rgba(0, 0, 0, 0.67);
        box-sizing: border-box;
        border-radius: 26px;
        display: flex;
        flex-wrap: wrap;

        div.profile__element{
            width:50%;
            height:50%;
          
            font-family: $primary-font;
            font-size: 30px;           
        }
    }

    hr.end__profile{
        width:85%;
        border: 1px solid rgba(0, 0, 0, 0.71);
    }

};