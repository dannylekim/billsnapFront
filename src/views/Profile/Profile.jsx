import React from "react";
import "./styles.scss";
import { FaPencilAlt, FaPhone, FaEnvelope, FaMapMarkerAlt, FaBirthdayCake } from "react-icons/fa";

const capilizeFirstLetter = (name) =>
  name.charAt(0).toUpperCase() + name.slice(1);

export default ({ userInfo }) => {
  const {
    firstName,
    middleName,
    lastName,
    phoneNumber,
    birthDate,
    location,
    email,
  } = userInfo;
  const userFullName = `${capilizeFirstLetter(firstName)} ${capilizeFirstLetter(
    middleName
  )} ${capilizeFirstLetter(lastName)}`; 

  return (
    <div className="profile__section">
      <div className="header__profile"> </div> 

      <img className="profile__avatar" src="./billSnapIcon.png" alt="avatar" />
      <div className="user__name">
        <h1> {userFullName} </h1>
        <button className="edit__profile">
          <FaPencilAlt />
        </button>
      </div>

      <div className="profile__info">
        <div className="profile__element"> <span> <FaPhone/> </span> {phoneNumber} </div>
        <div className="profile__element"> <span> <FaMapMarkerAlt/> </span> {`${capilizeFirstLetter(location.city)}, ${capilizeFirstLetter(location.country)}`} </div>
        <div className="profile__element"> <span> <FaEnvelope/> </span> {email} </div>
        <div className="profile__element"> <span> <FaBirthdayCake/> </span> {birthDate} </div>
      </div>
      <hr className="end__profile"/>

    </div>
  );
};