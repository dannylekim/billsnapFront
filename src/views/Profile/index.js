import Profile from "./Profile";
import {connect} from "react-redux";

const mapStateToProps = (state) => ({
  userInfo: state.users.userInfo,
});

export default connect(mapStateToProps)(Profile);
