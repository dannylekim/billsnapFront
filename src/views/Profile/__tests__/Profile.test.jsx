import React from "react";
import Profile from "../../Profile";
import { mockUserInfo } from "../constant/mockObjects";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureStore();

const store = mockStore({
  users: { userInfo: mockUserInfo },
});

const store2 = mockStore({
  users: { userInfo: {} },
});

describe("Profile", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("Profile should match snap shot when no users", () => {
        matches(
          <Provider store={store2}>
            <Profile
              userInfo={{ userInfo: {} }}
              hasUsers={false}
              history={{ push: jest.fn() }}
            />
          </Provider>
        );
      });

      it("Profile should match snap shot", () => {
        matches(
          <Provider store={store}>
            <Profile
              userInfo={mockUserInfo}
              hasUsers={true}
              history={{ push: jest.fn() }}
            />
          </Provider>
        );
      });
    });
  });
});
