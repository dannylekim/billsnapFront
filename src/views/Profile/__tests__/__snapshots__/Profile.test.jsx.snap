// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render snapshots ðŸ“¸ Profile should match snap shot 1`] = `
<div
  className="profile__section"
>
  <div
    className="header__profile"
  >
     
  </div>
  <img
    alt="avatar"
    className="profile__avatar"
    src="./billSnapIcon.png"
  />
  <div
    className="user__name"
  >
    <h1>
       
      String String String
       
    </h1>
    <button
      className="edit__profile"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z"
        />
      </svg>
    </button>
  </div>
  <div
    className="profile__info"
  >
    <div
      className="profile__element"
    >
       
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M493.4 24.6l-104-24c-11.3-2.6-22.9 3.3-27.5 13.9l-48 112c-4.2 9.8-1.4 21.3 6.9 28l60.6 49.6c-36 76.7-98.9 140.5-177.2 177.2l-49.6-60.6c-6.8-8.3-18.2-11.1-28-6.9l-112 48C3.9 366.5-2 378.1.6 389.4l24 104C27.1 504.2 36.7 512 48 512c256.1 0 464-207.5 464-464 0-11.2-7.7-20.9-18.6-23.4z"
          />
        </svg>
         
      </span>
       
      string
       
    </div>
    <div
      className="profile__element"
    >
       
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          />
        </svg>
         
      </span>
       
      String, String
       
    </div>
    <div
      className="profile__element"
    >
       
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
          />
        </svg>
         
      </span>
       
      string
       
    </div>
    <div
      className="profile__element"
    >
       
      <span>
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 384c-28.02 0-31.26-32-74.5-32-43.43 0-46.825 32-74.75 32-27.695 0-31.454-32-74.75-32-42.842 0-47.218 32-74.5 32-28.148 0-31.202-32-74.75-32-43.547 0-46.653 32-74.75 32v-80c0-26.5 21.5-48 48-48h16V112h64v144h64V112h64v144h64V112h64v144h16c26.5 0 48 21.5 48 48v80zm0 128H0v-96c43.356 0 46.767-32 74.75-32 27.951 0 31.253 32 74.75 32 42.843 0 47.217-32 74.5-32 28.148 0 31.201 32 74.75 32 43.357 0 46.767-32 74.75-32 27.488 0 31.252 32 74.5 32v96zM96 96c-17.75 0-32-14.25-32-32 0-31 32-23 32-64 12 0 32 29.5 32 56s-14.25 40-32 40zm128 0c-17.75 0-32-14.25-32-32 0-31 32-23 32-64 12 0 32 29.5 32 56s-14.25 40-32 40zm128 0c-17.75 0-32-14.25-32-32 0-31 32-23 32-64 12 0 32 29.5 32 56s-14.25 40-32 40z"
          />
        </svg>
         
      </span>
       
      string
       
    </div>
  </div>
  <hr
    className="end__profile"
  />
</div>
`;
